### PR TITLE
DAOS-9515 test: Add container query with property test

### PR DESCRIPTION
**Pull request by shimizukko**
Verify container properties are set during container query
over pydaos API.

Test Steps:
1. Create a container with some properties related to checksum, type, etc.
configured.
2. Call container query. it returns container info such as UUID, snapshots, etc.,
but not properties. We verify the property by passing in an empty data structure
as pass by reference.
3. Verify the property data structure passed in during step 2.

Added daos_property parameter in query() and fixed some typo.

Tag - Test
query_properties - container/query_properties.py
basic_checksum_object - checksum/csum_basic.py
containercreate - container/create.py
contdelete - contaienr/delete.py
cont_create_async - container/async.py
closehandle - container/open_close.py
snapshots - container/snapshot.py
dkey_akey_enum_punch - telemetry/dkey_akey_enum_punch.py

Skip-unit-tests: true
Test-tag: query_properties basic_checksum_object containercreate contdelete cont_create_async closehandle snapshots dkey_akey_enum_punch

Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

File: src/client/pydaos/raw/daos_api.py [hunk 1]
```diff
@@ -1663,19 +1662,37 @@ def close(self, coh=None, cb_func=None):
...
+                stores the property values. The values are set during create(). The
+                configuration of the data structure depends on the property values set.
+                See create() for more details. Defaults to None.
+            cb_func (CallbackHandler, None): Callback function. Defaults to None.
+
+        Returns:
+            daos_cref.ContInfo: Container info that stores UUID, snapshots, redundancy
+                factor, etc.
+
+        """
```
Comment (daltonbohning) [inline src/client/pydaos/raw/daos_api.py:1680]: A few comments:
1. It seems this is actually a `DaosProperty`, not `DaosContProperties`, right?
2. The C API is
    ```
    int
    daos_cont_query(daos_handle_t coh, daos_cont_info_t *info,
                    daos_prop_t *cont_prop, daos_event_t *ev);
    ```
    If it's not too much trouble, could we rename `daos_property` more similar to the C API? I.e. `cont_prop`
3. I'm not familiar with the pydaos API, but do we need to support backwards compatibility with the function definitions? Inserting a new param before `cb_func` isn't backwards compatible if someone uses positional arguments.

File: src/tests/ftest/container/query_properties.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            errors.append(msg)
+
+        # Verify values set in daos_property.
+        # Verify layout type.
+        layout_type = daos_property.dpp_entries[0].dpe_val
+        if layout_type == DaosContPropEnum.DAOS_PROP_CO_LAYOUT_POSIX.value:
+            self.log.info("Layout type is POSIX as expected.")
+        else:
+            self.log.info("Layout type is NOT POSIX!")
+            errors.append("Layout type is not POSIX! %s", layout_type)
```
Comment (daltonbohning) [inline src/tests/ftest/container/query_properties.py:118]: Any specific reason why we call `self.log.info` for layout type but not the other properties below?

**Review by daltonbohning — COMMENTED** [review]
Some files need the copyright year updated to 2022:
- `raw/daos_api.py`
- `server_utils.py`

Would it be reasonable to add the properties to the yaml so they can be configured? Or does that complicate the test too much?

File: src/client/pydaos/raw/daos_api.py (hunk 1 above)
Comment (shimizukko) [reply src/client/pydaos/raw/daos_api.py]: Thanks for the comment. I updated the variable name and moved the properties to the yaml.
For 3, I searched and noticed that no one uses positional arguments, and I ran those tests that calls `query()`, so I believe it's fine.

File: src/tests/ftest/container/query_properties.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/container/query_properties.py]: No, these are just for debugging.

File: src/client/pydaos/raw/daos_api.py (hunk 1 above)
Comment (daltonbohning) [reply src/client/pydaos/raw/daos_api.py]: We expose the `pydaos` API to users, but I'm not sure if the `pydaos.raw` API is expected to be backwards compatible:
https://docs.daos.io/v2.0/user/python/#Native_Programming_Interface

Comment (daltonbohning) [reply src/client/pydaos/raw/daos_api.py]: It might be okay, since I think the `pydaos.raw` API is primarily for our internal testing

**Review by daltonbohning — APPROVED** [review]

File: src/client/pydaos/raw/daos_api.py (hunk 1 above)
Comment (shimizukko) [reply src/client/pydaos/raw/daos_api.py]: That's a good point, but I noticed that `cb_func` comes last in all other methods, so I think it's better to be consistent. In addition, I think we should refactor this API if we expect external users to use it.

File: src/tests/ftest/container/query_properties.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        expected_layout_type = DaosContPropEnum.DAOS_PROP_CO_LAYOUT_POSIX.value
+        if actual_layout_type != expected_layout_type:
+            msg = "Layout type is not POSIX! Expected = {}; Actual = {}".format(
+                expected_layout_type, actual_layout_type)
+            errors.append(msg)
+
+        # Verify checksum.
+        if cont_prop.dpp_entries[1].dpe_val != chksum_type_exp:
+            msg = "Unexpected checksum from query! Expected = 1; Actual = {}".format(
+                cont_prop.dpp_entries[1].dpe_val)
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/container/query_properties.py]: Done. Thanks

File: src/client/pydaos/raw/daos_api.py (hunk 1 above)
Comment (daltonbohning) [reply src/client/pydaos/raw/daos_api.py]: Fair enough. I agree it needs work before external users use it.

**Review by dinghwah — APPROVED** [review]
LGTM

**Pull request merged**